File: dotnet/MoHandler/MoHandler.cs
 
 /*
 *   (C) Copyright 2009-2010 hSenid Software International (Pvt) Limited.
 *   All Rights Reserved.
 *
 *   These materials are unpublished, proprietary, confidential source code of
 *   hSenid Software International (Pvt) Limited and constitute a TRADE SECRET
 *   of hSenid Software International (Pvt) Limited.
 *
 *   hSenid Software International (Pvt) Limited retains all title to and intellectual
 *   property rights in these materials.
 */
using System;
using System.Collections.Generic;
using System.Text;
using System.Web;

namespace MoHandler
{
    public class MoHandler : IHttpHandler
    {       
        public void ProcessRequest(HttpContext context)
        {
            HttpRequest Request = context.Request;
            HttpResponse Response = context.Response;           
            Response.Write("<html>");
            Response.Write("<body>");
            Response.Write("<h1> Hello from MO handler. </h1>");
            Response.Write("<h1> correlator is :" + Request.Form.Get("correlator").ToString() + "</h1>");
            Response.Write("<h1> addresss is : " +Request.Form.Get("address").ToString() + "</h1>");
            Response.Write("<h1> message  is : " + Request.Form.Get("message").ToString() + "</h1>");
            Response.Write("</body>");
            Response.Write("</html>");         
        }
        public bool IsReusable
        {
            get { return false; }
        }      
            
    }
}

File: ruby/mo-sample/lib/sampleapp/MoHandler.rb
#/*
# *   (C) Copyright 2010-2011 hSenid Software International (Pvt) Limited.
# *   All Rights Reserved.
# *
# *   These materials are unpublished, proprietary, confidential source code of
# *   hSenid Software International (Pvt) Limited and constitute a TRADE SECRET
# *   of hSenid Software International (Pvt) Limited.
# *
# *   hSenid Software International (Pvt) Limited retains all title to and intellectual
# *   property rights in these materials.
# *
# */


#!/sw/bin/ruby
require 'webrick'
require 'net/http'
require 'uri'
include WEBrick

#  $LastChangedDate$
#  $LastChangedBy$
#  $LastChangedRevision$

class MoHandler  < WEBrick::HTTPServlet::AbstractServlet
   def do_POST(request, response)
      response.body = "<HTML>" + "\n"
      response.body << "<Head>" + "\n"
      response.body << "message is : " + request.query['message'] + "\n"
      response.body << "address is : " + request.query['address'] + "\n"
      response.body << "correlator is : " + request.query['correlator'] + "\n"
      response.body << "</Head>" + "\n"
      response.body << "</HTML>" + "\n"
   end
end

# ----------------------------------------------
# Create an HTTP server
s = HTTPServer.new(
  :Port            => 8002
)
s.mount("/MoHandler", MoHandler)
# When the server gets a control-C, kill it
trap("INT"){ s.shutdown }
# Start the server
s.start
s.listeners